cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1480,7 +1486,51 @@ func maybeMountSave(disk disks.Disk, rootdir string, encrypted bool, mountOpts *
 	return true, nil
 }
 
+func populateSysroot(mst *initramfsMountsState) error {
```

Comment:
Could not we create a file in `/run` and use `ConditionPathExists=` to choose what set of `.mount` units to use?

Using systemd units allow to have different ways to mount things depending on the version. Having it static in snap-boostrap makes it more rigid.

Reply:
About using `Cond*` we have the problem about detection of UC/classic, see above.

Regarding dependencies what depends on sysroot mounts depends also on `snap-initramfs-mounts.service`, so it should not be a problem, although I'm not 100% sure tbh. But note that we also mount other bits on snap-bootstrap and other services depend on that too.